### ISSUE TITLE
SDL OpenGL header license is non-free (SGI Free B v1.1)

### ISSUE THREAD

**Issue by Conan-Kudo** [libsdl-org/sdl12-compat#105]
During a review, it was discovered that there is some code that's listed under the terms of the [SGI Free B v1.1 license](https://spdx.org/licenses/SGI-B-1.1.html):

https://github.com/libsdl-org/sdl12-compat/blob/c0504eb42c7cabc23c6b4a16f5915ee49962ddff/include/SDL/SDL_opengl.h#L1-L30

This license is non-free, but it's easy enough to fix as the license has a clause to allow upgrading to the [SGI Free B v2.0 license (which is effectively MIT)](https://spdx.org/licenses/SGI-B-2.0.html).

Can the license terms be updated accordingly?

**Comment by sezero** [comment]
Since it's upgradeable, I guess it can/should be fixed.

I wonder though, why this was not a problem with SDL-1.2 itself, because the
header is a direct copy of the SDL-1.2 version.

**Comment by Conan-Kudo** [comment]
It probably was but nobody noticed.

**Comment by sezero** [comment]
@icculus, @slouken ?

**Comment by icculus** [comment]
Yeah, I took this from sdl 1.2 directly, but we can upgrade it.

**Comment by sezero** [comment]
What is the patch for this, just replace the SGI licence notice?

**Comment by Conan-Kudo** [comment]
Well, it's updating the license notice to indicate that SDL has upgraded it to Free B 2.0

**Comment by Conan-Kudo** [comment]
Though @spotrh was the one that handled that stuff for Xorg and Mesa long ago, he might have better advice...

**Comment by spotrh** [comment]
Ah yes, SGI Free B, my old enemy. If you look at Section 8 of SGI Free B 1.1 (the very non-free version), it says:

`8. Versions of the License. SGI may publish revised and/or new versions of the License from time to time, each with a distinguishing version number. Once Covered Code has been published under a particular version of the License, Recipient may, for the duration of the license, continue to use it under the terms of that version, or choose to use such Covered Code under the terms of any subsequent version published by SGI. `

In 2008, SGI (after much prodding and poking) released the SGI Free B 2.0 license, which is MIT:
https://spdx.org/licenses/SGI-B-2.0.html

Because of section 8 in 1.1, anyone using code under 1.1 can choose to use it under 2.0. The proper way to do this is to update the license notice to indicate that you are using code under the terms of the SGI Free B 2.0 license, and also include a copy of the SGI Free B 2.0 license text (since it is just MIT and MIT requires that "the above copyright notice ... shall be included in all copies or substantial portions of the Software."). I recommend that you include this license text as a separate file, since it makes it easier for binary distributors to comply with the license.

TLDR: sed 's/SGI Free Software License B, Version 1.1/SGI Free Software License B, Version 2.0/g' && vi LICENSE-SGI-Free-B-2.0.txt

**Comment by icculus** [comment]
I'm wondering if we could just pull in something newer from Khronos and dump the SGI license entirely. Not sure if that's possible or if it causes new problems, though.

**Comment by sezero** [comment]
> I'm wondering if we could just pull in something newer from Khronos and dump the SGI license entirely. Not sure if that's possible or if it causes new problems, though.

In that case, we need upgrade to at least v39 with new Khronos
license/copyright from Feb 26, 2007.

- We are currently at v29, here: https://github.com/mesa3d/mesa/commit/c8f8fbecd7e3b858d1fcbe64bb0ed3e2295f2d5d#diff-ebb5846fcbbbca04156cbb46520639f1a3221f2403eee0f4432bb2d9146d2829
- mesa updates to v34: https://github.com/mesa3d/mesa/commit/3cd07d2c390099ca0f9b2e27a548920f880715c4#diff-ebb5846fcbbbca04156cbb46520639f1a3221f2403eee0f4432bb2d9146d2829
- mesa adds #ifdef tests for Windows for int64_t, uint64_t: https://github.com/mesa3d/mesa/commit/a3a640171f12de91202b5dc4777ffd53be10a0aa#diff-ebb5846fcbbbca04156cbb46520639f1a3221f2403eee0f4432bb2d9146d2829
- mesa updates to v39: https://github.com/mesa3d/mesa/commit/c226e44437ed13149046d5b6eef6662b52e79848#diff-ebb5846fcbbbca04156cbb46520639f1a3221f2403eee0f4432bb2d9146d2829
- mesa updates version with new Khronos license/copyright: https://github.com/mesa3d/mesa/commit/344c3f7379ab4db749e7e8513be9be8fbb9f126c#diff-ebb5846fcbbbca04156cbb46520639f1a3221f2403eee0f4432bb2d9146d2829

I _think_ (not actually tested) there won't be problems if we apply our own 
customizations correctly.

**Comment by sezero** [comment]
And for convenience, here is the diff between original v29 glext.h and
our own SDL1.2/include/SDL_opengl.h:  [glext_h.diff.txt](https://github.com/libsdl-org/sdl12-compat/files/6673742/glext_h.diff.txt)

**Comment by sezero** [comment]
Here is an SDL_opengl.h based on v39 mesa glext.h from Mar 25, 2008, i.e. this one:
https://github.com/mesa3d/mesa/blob/7048545c44ccc5a2ffe716705c128d7a37dff766/include/GL/glext.h
[SDL_opengl.h.zip](https://github.com/libsdl-org/sdl12-compat/files/6674012/SDL_opengl.h.zip)

Here is the diff between original glext.h and this SDL_opengl.h (read carefully):
[glext.h-v39-to-SDL_opengl.h.txt](https://github.com/libsdl-org/sdl12-compat/files/6674020/glext.h-v39-to-SDL_opengl.h.txt)

IS this OK to apply to SDL-1.2?  If it is, I will push it to SDL-1.2 and then apply the
minor change between sdl12-compat version and push here.

**Comment by sezero** [comment]
And here is the sdl12-compat version of the above header:  [SDL_opengl.h-for-sdl12compat.zip](https://github.com/libsdl-org/sdl12-compat/files/6674075/SDL_opengl.h-for-sdl12compat.zip)

**Comment by sezero** [comment]
Revised versions fixing gotchas related to `GLEXT_64_TYPES_DEFINED` stuff.
[SDL_opengl.h-for-SDL-1.2.zip](https://github.com/libsdl-org/sdl12-compat/files/6677237/SDL_opengl.h-for-SDL-1.2.zip)
[glext.h-v39-to-SDL_opengl.h.txt](https://github.com/libsdl-org/sdl12-compat/files/6677239/glext.h-v39-to-SDL_opengl.h.txt)
[SDL_opengl.h-for-sdl12compat.zip](https://github.com/libsdl-org/sdl12-compat/files/6677240/SDL_opengl.h-for-sdl12compat.zip)

OK?

**Comment by icculus** [comment]
This looks okay to me.

**Comment by sezero** [comment]
Fixed as of https://github.com/libsdl-org/sdl12-compat/commit/0da355b56d41477b653d3d76d75b7918c89afdf1

**Comment by sezero** [comment]
We can actually upgrade further to v86 of glext.h. Among _a lot_ of
other things, it has fixes to constants such as `GL_ELEMENT_ARRAY_APPLE`,
`GL_ELEMENT_ARRAY_TYPE_APPLE`, and `GL_ELEMENT_ARRAY_POINTER_APPLE`
(those particular three hadn't been fixed until v61.)

I don't know whether it's worth it, though.

**Comment by icculus** [comment]
It's not worth it.
